### DISCUSSION TITLE
Help with automatic updates for MAUI desktop app

### DISCUSSION THREAD

**Discussion by muheebthewizard** [dotnet/maui#15255] [Q&A]
Hello everyone,
I have developed a .net maui desktop app for a client. Now I want to distribute it to them ad-hoc. So I am publishing a msix package via visual studio. And I am trying to enable automatic updates by selecting the option. 
What happens here is after I tried a use a url, the copy and close doesn't appear,which is supposed to copy the installer file to that location so my client can get updates. Then I tried to create a local unc path and use that to test automatic updates. Now the copy and close option appears, but after clicking it, I checked the location and there are no files there at all.

So, please help me understand how to provide automatic updates for maui desktop app without using windows store option.

**Comment by Xyncgas** [comment]
Malwares use a mechanism where they fetch the files from the server to change their behavior, if that's desirable you start with defining your service area, then have that part of program to be dynamically loaded
Either by replacing dll so you are dynamically loading them in your program, or seperate launcher with client like many games do, or ship a mini business engine / virtual machine that interpret some kinds of script that let the program do this or that (also commonly implemented inside game)
